### PR TITLE
feat(Outlook): add presence

### PR DESCRIPTION
**Pull request by kyrie25**
#### Resolves discussion #5179 

<details>
<summary> Proof of the creation working: </summary>
<br>

image
image
image
image
image
image
image

</details>

### CODE REVIEW

File: websites/O/Outlook/presence.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    if (
+      document.querySelector(
+        "#ReadingPaneContainerId > div > div > div > div:nth-child(1) > div._3Ot6xv41uIO58lh-I36wdt > div:nth-child(1) > div > div._1LtJxmUY1w2weHRM-NvCf9 > div"
+      ) ||
+      path.includes("compose")
+    )
+      presenceData.details = "Composing a mail";
+    else if (path.includes("id")) {
+      presenceData.details = "Reading a mail";
+      if (await presence.getSetting("title")) {
```
Comment (EncryptedDev) [inline websites/O/Outlook/presence.ts:22]: Missing type 
```suggestion
      if (await presence.getSetting<boolean>("title")) {
```

**Review by EncryptedDev — CHANGES_REQUESTED** [review]

File: websites/O/Outlook/presence.ts (hunk 1 above)
Comment (kyrie25) [reply websites/O/Outlook/presence.ts]: done

**Review by EncryptedDev — CHANGES_REQUESTED** [review]
Use `email` instead of `mail` in the strings. 

```diff
- Viewing sent mails
- Viewing junk mails
+ Viewing sent emails
+ Viewing junk emails
```

**Comment by kyrie25** [comment]
Oh I used the `Gmail` presence as reference for the `.details` so thought those were appropriate. Should I continue changing them @EncryptedDev

**Comment by EncryptedDev** [comment]
> Oh I used the `Gmail` presence as reference for the `.details` so thought those were appropriate. Should I continue changing them @EncryptedDev

I figured. I was actually in the process of rewriting the Gmail presence xD

**Review by EncryptedDev — COMMENTED** [review]
pref: assign types to querySelectors? you do it in all your other presences

File: websites/O/Outlook/presence.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+      path.includes("compose")
+    )
+      presenceData.details = "Composing an email";
+    else if (path.includes("id")) {
+      presenceData.details = "Reading an email";
+      if (await presence.getSetting<boolean>("title")) {
+        presenceData.state = document.querySelector(
+          "#ReadingPaneContainerId div._2bnn4NUZa-NanNIO4GItP0.allowTextSelection._3FNHkYLZYD6Y3-QNc7ZBo2 > span"
+        ).textContent;
+      } else presenceData.state;
```
Comment (EncryptedDev) [inline websites/O/Outlook/presence.ts:26]: This is useless 
```suggestion
      }
```

File: websites/O/Outlook/presence.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+      presenceData.details = "Viewing sent emails";
+    else if (path.includes("conversationhistory"))
+      presenceData.details = "Viewing conversation history";
+    else presenceData.details = "Browsing emails";
+  } else if (path.startsWith("/calendar"))
+    presenceData.details = "Viewing calendar";
+  else if (path.startsWith("/files")) presenceData.details = "Browsing files";
+  else if (path.startsWith("/people"))
+    presenceData.details = "Viewing contact list";
+  else presenceData;
```
Comment (EncryptedDev) [inline websites/O/Outlook/presence.ts:42]:
```suggestion
```

**Review by EncryptedDev — CHANGES_REQUESTED** [review]

**Comment by kyrie25** [comment]
> pref: assign types to querySelectors? you do it in all your other presences

Oh yeah I forgot about that, will do ASAP

**Review by EncryptedDev — APPROVED** [review]
lgtm

**Review by Slowlife01 — APPROVED** [review]

**Pull request merged**